### PR TITLE
Add support for media and node revisions when updating

### PR DESCRIPTION
**Pull request by hassanelsheikha**
Hi @mjordan. This PR is a prerequisite for [this PR](https://github.com/mjordan/islandora_workbench/pull/645) for creating new revision log messages during the `update` and `update_media` routines. It adds two hooks for creating the new revisions upon node/media saves.

### CODE REVIEW

File: config/optional/config/optional/rest.resource.entity.node_type.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+id: entity.node_type
+plugin_id: 'entity:node_type'
+granularity: resource
+configuration:
+  methods:
+    - GET
+  formats:
+    - json
+  authentication:
+    - basic_auth
```
Comment (mjordan) [inline config/optional/config/optional/rest.resource.entity.node_type.yml:18]: This file is a duplicate and should be deleted.

Comment (hassanelsheikha) [reply config/optional/config/optional/rest.resource.entity.node_type.yml]: Done! Sorry about that.

Comment (hassanelsheikha) [reply config/optional/config/optional/rest.resource.entity.node_type.yml]: Sorry one sec... GitHub is being weird

Comment (hassanelsheikha) [reply config/optional/config/optional/rest.resource.entity.node_type.yml]: Ok I think I got it. Please confirm

**Comment by mjordan** [comment]
Yes, I think that's correct.

**Pull request merged**